spec/system/homepage_spec.rb

```diff
@@ -10,11 +10,23 @@
     expect(page).not_to have_css(".homepage-inverse-header__title")
   end
 
-  it "renders the search with autocomplete component with the correct source URL" do
-    visit "/"
+  context "search autocomplete" do
+    it "renders the search with autocomplete component with the correct source URL when autocomplete is enabled" do
+      ClimateControl.modify GOVUK_DISABLE_SEARCH_AUTOCOMPLETE: nil do
+        visit "/"
+
+        expect(page).to have_css(".gem-c-search-with-autocomplete")
+        expect(page).to have_css("[data-source-url='http://www.dev.gov.uk/api/search/autocomplete.json']")
+      end
+    end
 
-    expect(page).to have_css(".gem-c-search-with-autocomplete")
-    expect(page).to have_css("[data-source-url='http://www.dev.gov.uk/api/search/autocomplete.json']")
+    it "does not render the search with autocomplete component when autocomplete is disabled" do
+      ClimateControl.modify GOVUK_DISABLE_SEARCH_AUTOCOMPLETE: "1" do
+        visit "/"
+
+        expect(page).to_not have_css(".gem-c-search-with-autocomplete")
```

Comment:
I imagine it'd be good to also assert that we have `.gem-c-search` to flag that it renders search component instead.